src/Compilers/Core/Portable/MetadataReference/Metadata.cs

```diff
@@ -14,6 +14,11 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     public sealed class MetadataId
     {
+        private MetadataId()
+        {
+        }
+
+        internal static MetadataId CreateNewId() => new MetadataId();
```

Comment:
@AlekseyTs Was just matching the pattern in the IDE code for our ID types.  As it's internal though, it shouldn't really matter one way or the other.  But i thought i'd at least be consistent there :)